system_tests_qdmanage failure on CI

```
23: =================================== FAILURES ===================================
23: ___________________________ QdmanageTest.test_query ____________________________
23: 
23: self = <system_tests_qdmanage.QdmanageTest testMethod=test_query>
23: 
23:     def test_query(self):
23:     
23:         def long_type(name):
23:             return 'org.apache.qpid.dispatch.' + name
23:     
23:         types = ['listener', 'log', 'router']
23:         long_types = [long_type(name) for name in types]
23:     
23:         qall = json.loads(self.run_qdmanage('query'))
23:         qall_types = {e['type'] for e in qall}
23:         for t in long_types:
23:             self.assertIn(t, qall_types)
23:     
23:         qlistener = json.loads(self.run_qdmanage('query --type=listener'))
23:         self.assertEqual([long_type('listener')] * 4, [e['type'] for e in qlistener])
23:         self.assertEqual(self.router_1.ports[0], int(qlistener[0]['port']))
23:     
23:         qattr = json.loads(self.run_qdmanage('query type name'))
23:         for e in qattr:
23:             self.assertEqual(2, len(e))
23:     
23:         def name_type(entities):
23:             ignore_types = [long_type(t) for t in ['router.link', 'connection', 'router.address']]
23:             return set((e['name'], e['type']) for e in entities
23:                        if e['type'] not in ignore_types)
23: >       self.assertEqual(name_type(qall), name_type(qattr))
23: E       AssertionError: Items in the first set but not the second:
23: E       ('router.node/R2', 'org.apache.qpid.dispatch.router.node')
23: 
23: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_qdmanage.py:190: AssertionError
```
https://github.com/skupperproject/skupper-router/runs/5557333167?check_suite_focus=true#step:9:4219

Duplicates https://github.com/skupperproject/skupper-router/issues/143

Current error is slightly different, given the skupper renames

```
23: ___________________________ SkmanageTest.test_query ____________________________
23: 
23: self = <system_tests_skmanage.SkmanageTest testMethod=test_query>
23: 
23:     def test_query(self):
23:     
23:         def long_type(name):
23:             return 'io.skupper.router.' + name
23:     
23:         types = ['listener', 'log', 'router']
23:         long_types = [long_type(name) for name in types]
23:     
23:         qall = json.loads(self.run_skmanage('query'))
23:         qall_types = {e['type'] for e in qall}
23:         for t in long_types:
23:             self.assertIn(t, qall_types)
23:     
23:         qlistener = json.loads(self.run_skmanage('query --type=listener'))
23:         self.assertEqual([long_type('listener')] * 4, [e['type'] for e in qlistener])
23:         self.assertEqual(self.router_1.ports[0], int(qlistener[0]['port']))
23:     
23:         qattr = json.loads(self.run_skmanage('query type name'))
23:         for e in qattr:
23:             self.assertEqual(2, len(e))
23:     
23:         def name_type(entities):
23:             ignore_types = [long_type(t) for t in ['router.link', 'connection', 'router.address']]
23:             return set((e['name'], e['type']) for e in entities
23:                        if e['type'] not in ignore_types)
23: >       self.assertEqual(name_type(qall), name_type(qattr))
23: E       AssertionError: Items in the first set but not the second:
23: E       ('router.node/R2', 'io.skupper.router.router.node')
23: 
23: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_skmanage.py:190: AssertionError
23: - generated xml file: /__w/skupper-router/skupper-router/skupper-router/build/tests/junitxmls/system_tests_skmanage.xml -
23: =========================== short test summary info ============================
23: FAILED ::SkmanageTest::test_query - AssertionError: Items in the first set bu...
23: =================== 1 failed, 30 passed in 199.06s (0:03:19) ===================
31/32 Test #23: system_tests_skmanage ................***Failed  201.02 sec
```

--- a/tests/system_tests_skmanage.py
+++ b/tests/system_tests_skmanage.py
@@ -179,15 +179,24 @@ class SkmanageTest(TestCase):
         self.assertEqual([long_type('listener')] * 4, [e['type'] for e in qlistener])
         self.assertEqual(self.router_1.ports[0], int(qlistener[0]['port']))
 
-        qattr = json.loads(self.run_skmanage('query type name'))
-        for e in qattr:
-            self.assertEqual(2, len(e))
-
         def name_type(entities):
             ignore_types = [long_type(t) for t in ['router.link', 'connection', 'router.address']]
             return set((e['name'], e['type']) for e in entities
                        if e['type'] not in ignore_types)
-        self.assertEqual(name_type(qall), name_type(qattr))
+
+        test_passed = False
+        for i in range(10):
+            qattr = json.loads(self.run_skmanage('query type name'))
+            try:
+                for e in qattr:
+                    self.assertEqual(2, len(e))
+                self.assertEqual(name_type(qall), name_type(qattr))
+                test_passed = True
+                break
+            except AssertionError as ae:
+                # Give some time for the Router 1's data structures to update.
+                sleep(0.5)
+        self.assertTrue(test_passed)
 
     def test_get_schema(self):
         schema = dictify(QdSchema().dump())

--- a/tests/system_tests_skmanage.py
+++ b/tests/system_tests_skmanage.py
@@ -28,7 +28,7 @@ from proton.utils import BlockingConnection
 from skupper_router_internal.compat import dictify
 from skupper_router_internal.management.qdrouter import QdSchema
 
-from system_test import unittest
+from system_test import unittest, retry_exception
 from system_test import Logger, TestCase, Process, Qdrouterd, main_module, TIMEOUT, DIR
 from system_test import QdManager
 
@@ -184,19 +184,13 @@ class SkmanageTest(TestCase):
             return set((e['name'], e['type']) for e in entities
                        if e['type'] not in ignore_types)
 
-        test_passed = False
-        for i in range(10):
+        def query_type_name():
             qattr = json.loads(self.run_skmanage('query type name'))
-            try:
-                for e in qattr:
-                    self.assertEqual(2, len(e))
-                self.assertEqual(name_type(qall), name_type(qattr))
-                test_passed = True
-                break
-            except AssertionError as ae:
-                # Give some time for the Router 1's data structures to update.
-                sleep(0.5)
-        self.assertTrue(test_passed)
+            for e in qattr:
+                self.assertEqual(2, len(e))
+            self.assertEqual(name_type(qall), name_type(qattr))
+
+        retry_exception(query_type_name)
 
     def test_get_schema(self):
         schema = dictify(QdSchema().dump())

